src/Mvc/test/WebSites/ControllersFromServicesWebSite/Startup.cs

```diff
@@ -32,10 +32,7 @@ public void ConfigureServices(IServiceCollection services)
                       typeof(ComponentFromServicesViewComponent),
                       typeof(InServicesTagHelper)));
 
-                    var relatedAssenbly = RelatedAssemblyAttribute
-                        .GetRelatedAssemblies(GetType().Assembly, throwOnError: true)
-                        .SingleOrDefault();
-                    foreach (var part in CompiledRazorAssemblyApplicationPartFactory.GetDefaultApplicationParts(relatedAssenbly))
+                    foreach (var part in CompiledRazorAssemblyApplicationPartFactory.GetDefaultApplicationParts(Assembly.GetExecutingAssembly()))
```

Comment:
Doesn't the consolidated part factory already take care of this?

Reply:
It does. However, this particularly startup logic removes all the discovered application parts and adds its own.

https://github.com/dotnet/aspnetcore/blob/35bb1b86e73c4ea147f2454aa0284e1c02dd9bdf/src/Mvc/test/WebSites/ControllersFromServicesWebSite/Startup.cs#L26

I think it does this to validate the behavior of the specific extension methods used here.